geoopt/manifolds/poincare/math.py

```diff
@@ -84,7 +84,7 @@ def _eps(x):
 
 
 def _project(x, c, dim: int = -1, eps: float = None):
-    norm = x.norm(dim=dim, keepdim=True, p=2)
+    norm = x.norm(dim=dim, keepdim=True, p=2).clamp_min(1e-15)
```

Comment:
Please introduce some sort of `EPS = 1e-15` even if it be package-level global constant